cmd/annotations-checker/validate-annotations.go

```diff
@@ -0,0 +1,119 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/google/go-github/v32/github" // GitHub API client library
+	// "github.com/ministryofjustice/cloud-platform-environments/pkg/namespace"
+	"golang.org/x/oauth2" // API requests authentication
+	// "google.golang.org/genproto/googleapis/api/annotations"
+	"gopkg.in/yaml.v2"
+)
+
+// Annotations structs to define the structure of annotations in the YAML file
+type Annotations struct {
+	Kind           string `yaml:"kind"`
+	SourceCodeRepo string `yaml:"cloud-platform.justice.gov.uk/source-code"`
+	TeamName       string `yaml:"cloud-platform.justice.gov.uk/team-name"`
+}
+
+// hold the structure for the YAML file
+type KubernetesConfig struct {
+	Metadata struct {
+		Annotations Annotations `yaml:"annotations"`
+	} `yaml:"metadata"`
+}
+
+// gitHub client
+func NewGitHubClient(token string) *github.Client {
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})
+	tc := oauth2.NewClient(ctx, ts)
+	client := github.NewClient(tc)
+	return client
+}
+
+// validate team name and source code repo
+func ValidateAnnotations(client *github.Client, org string) (string, error) {
+	// get namespace directory
+	namespacesDir := "/path/to/namespaces"
+	files, err := os.ReadDir(namespacesDir)
+	if err != nil {
+		return "", fmt.Errorf("Failed to read namespaces directory: %v", err)
+	}
+
+	// check if kind:Namespace exists in any file
+	foundNamespace := false
+	for _, file := range files {
```

Comment:
Currently, this will loop over every file in the namespace (so it will always trigger). We only want to check new files or files that have been modified. We don't care about files that already exist (we can address those later) and we don't care about files that are being deleted either.

If you struggle to figure out a way to do this for your current code consider calling a git diff on the PR against main or exploring passing down the `github.context` (specifically accessing the event) from the action [this](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) and [this](https://stackoverflow.com/a/77153325) link might be helpful.